src/persist-client/src/batch.rs

```diff
@@ -146,18 +152,18 @@ where
     /// marks them as deleted.
     #[instrument(level = "debug", skip_all, fields(shard = %self.shard_id))]
     pub async fn delete(mut self) {
-        // TODO: This is temporarily disabled because nemesis seems to have
-        // caught that we sometimes delete batches that are later needed.
-        // Temporarily removing the deletions while we figure out the bug in
-        // case it has anything to do with CI timeouts.
-        //
-        // for key in self.blob_keys.iter() {
-        //     retry_external("batch::delete", || async {
-        //         self.blob.delete(key).await
-        //     })
-        //     .await;
-        // }
-        self.batch.parts.clear();
+        if !self.batch_delete_enabled {
+            return;
+        }
+        let _handle = mz_ore::task::spawn(|| "persist::batch::delete", async move {
```

Comment:
Modest preference for:
- Doing the deletes directly, instead of spawning a task.
- Doing the deletes concurrently.

Both of those changes are probably slightly more aggressive, but they don't add extra data-loss risk, and given that this is behind a flag it seems pretty safe to me.

Reply:
Done and done!